tests/CMakeLists.txt

```diff
@@ -1,75 +1,49 @@
-cmake_minimum_required(VERSION 3.13)
-# https://github.com/google/oss-policies-info/blob/main/foundational-cxx-support-matrix.md#foundational-c-support
-
-project(test_libdisplaydevice)
-
-include_directories("${CMAKE_SOURCE_DIR}")
-
+#
+# Setup google test
+#
 enable_testing()
-
-# Add GoogleTest directory to the project
-set(GTEST_SOURCE_DIR "${CMAKE_SOURCE_DIR}/third-party/googletest")
 set(INSTALL_GTEST OFF)
 set(INSTALL_GMOCK OFF)
-add_subdirectory("${GTEST_SOURCE_DIR}" "${CMAKE_CURRENT_BINARY_DIR}/googletest")
-include_directories("${GTEST_SOURCE_DIR}/googletest/include" "${GTEST_SOURCE_DIR}")
-
-# coverage
+include(GoogleTest)
+add_subdirectory("${PROJECT_SOURCE_DIR}/third-party/googletest" "third-party/googletest")
+
+# A helper function to setup the test executables
+function(add_dd_tests)
+    set(options "")
+    set(oneValueArgs "")
+    set(multiValueArgs TESTS LIBRARIES)
+    cmake_parse_arguments(FN_VARS "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    foreach(test ${FN_VARS_TESTS})
+        add_executable(${test} ${test}.cpp)
+        target_link_libraries(${test}
+                gtest
+                gtest_main  # if we use this we don't need our own main function
+                libdisplaydevice # we are always testing at least the public API so it's safe to always link this
+                libfixtures # these are our fixtures/helpers for the tests
+                ${FN_VARS_LIBRARIES} # additional libraries if needed
+        )
+
+        # Adds the test to CTest
+        gtest_discover_tests(${test})
+    endforeach()
+endfunction()
+
+#
+# Add subdirectories
+#
+add_subdirectory(fixtures)
+add_subdirectory(unit)
+
+#
+# Additional setup for coverage
 # https://gcovr.com/en/stable/guide/compiling.html#compiler-options
+#
 set(CMAKE_CXX_FLAGS "-fprofile-arcs -ftest-coverage -O1")
 set(CMAKE_C_FLAGS "-fprofile-arcs -ftest-coverage -O1")
 
-# if windows
+# TODO: why is this needed?
```

Comment:
see here: https://google.github.io/googletest/quickstart-cmake.html#set-up-a-project

Why is it needed? I don't know. But it's in the most basic gtest example, so that's why it's there.